### PR TITLE
test: add integration test for plugins together

### PR DESCRIPTION
**Pull request by JamesHenry**
- Removes all the old integration tests from typescript-eslint-parser, they are not so useful any more now that our unit tests cover some fundamental integration points implicitly

- Adds a new way of doing the integration test using some docker magic and writing a snapshot back to the outside world ✨ 

- Adds an integration test for using both plugins together in the same project 😍 

https://github.com/typescript-eslint/typescript-eslint/blob/6cb1e04af20be21b7895b9ef8103e0ba272947c4/tests/integration/fixtures/typescript-and-tslint-plugins-together/test.js.snap

### CODE REVIEW

File: packages/eslint-plugin-typescript/package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
   },
   "dependencies": {
     "requireindex": "^1.2.0",
-    "@typescript-eslint/parser": "21.0.2"
+    "typescript-eslint-parser": "21.0.2"
```
Comment (armano2) [inline packages/eslint-plugin-typescript/package.json]: what is a reason of renaming package back to old name?

Comment (JamesHenry) [reply packages/eslint-plugin-typescript/package.json]: It causes issues during the installation in the integration tests

Comment (JamesHenry) [reply packages/eslint-plugin-typescript/package.json]: We can update all the names at once when we publish, it didn't really make sense to do a partial job of it anyway

**Comment by JamesHenry** [comment]
The second commit is just reverting renames, the first commit contains the relevant changes

**Pull request merged**